### PR TITLE
Fix NEON build issue: type marker instance creation

### PR DESCRIPTION
**Pull request by sshockwave**
Build error
```
/libjxl/lib/jxl/fast_dct-inl.h:162:64: error: expected '(' for function-style cast or type construction
      N * M * 2 + 3 * std::max(N, M) * MaxLanes(HWY_FULL(float)));
```
due to `HWY_FULL(float)` (aka `ScalableTag<float>`) being a type.

Related: 4b8aad231bc7857158be8e4bf554a805596b1916

### CODE REVIEW

**Comment by mo271** [comment]
Thanks for the fix!
Please add your name to the AUTHORS file.

**Comment by sshockwave** [comment]
@mo271 Done!

**Comment by mo271** [comment]
@sshockwave Thanks!
Our check still does not pass, because the name used as the author in the commit, see 
https://github.com/libjxl/libjxl/commit/9b03f99e199735ecf129e68f45c76c8e01420eb5.patch
does not match the name in the AUTHORS file...

Could you either edit the AUTHORS file or change the author of the commits?

**Comment by sshockwave** [comment]
@mo271 I see, I just updated the AUTHORS list.

**Comment by eustas** [comment]
Better fix would be to extract `const HWY_FULL(float) df;`.

**Comment by sshockwave** [comment]
@eustas Why create a dummy variable? Could you explain more, please? Thanks!

**Comment by eustas** [comment]
Mostly for readability. Out of 132 places where HWY_FULL is used only 7 are not variable declaration. Likely we should convert those 7 as well.

**Comment by sshockwave** [comment]
Understood. I'll make the change.

**Review by mo271 — APPROVED** [review]

**Comment by mo271** [comment]
After merging #2755, we have a mix of `const` and `consexpr`. Should it make it more consistent, @eustas?

**Comment by sshockwave** [comment]
Seems that `const HWY_FULL(T) d;` is automatically `constexpr`. I'm OK with both.

File: lib/extras/metrics.cc [hunk 1]
```diff
@@ -41,7 +41,7 @@ double ComputeDistanceP(const ImageF& distmap, const ButteraugliParams& params,
 #else
     using T = float;
 #endif
-    const HWY_FULL(T) d;
+    constexpr HWY_FULL(T) d;
```
Comment (mo271) [inline lib/extras/metrics.cc:44]:
```suggestion
    const HWY_FULL(T) d;
```

File: lib/jxl/fast_dct-inl.h [hunk 2]
```diff
@@ -158,8 +158,9 @@ HWY_NOINLINE void TestFastIDCT() {
   auto idct_mem = hwy::AllocateAligned<int16_t>(N * M);
   int16_t* idct = idct_mem.get();
 
+  constexpr HWY_FULL(float) df;
```
Comment (mo271) [inline lib/jxl/fast_dct-inl.h:161]:
```suggestion
  const HWY_FULL(float) df;
```

File: lib/jxl/gauss_blur.cc [hunk 3]
```diff
@@ -381,7 +381,7 @@ void FastGaussianVertical(const hwy::AlignedUniquePtr<RecursiveGaussian>& rg,
                           ImageF* JXL_RESTRICT out) {
   JXL_CHECK(SameSize(in, *out));
 
-  const HWY_FULL(float) df;
+  constexpr HWY_FULL(float) df;
```
Comment (mo271) [inline lib/jxl/gauss_blur.cc:384]:
```suggestion
  const HWY_FULL(float) df;
```

**Comment by mo271** [comment]
ok, great! Then let's make it `const` in all places...

**Review by mo271 — APPROVED** [review]

**Comment by mo271** [comment]
Thanks for the contribution!

**Pull request merged**